request-example.go

```diff
@@ -151,15 +170,38 @@ func (fs *root) Filecmd(r *Request) error {
 
 	switch r.Method {
 	case "Setstat":
+		// We explicitly do not support changing the mode of symlinks.
+		// Nor do we want to change the mode of whatever the symlink is pointed
+		// at.
+		link, err := fs.Readlink(r.Filepath)
+		if link != "" || err == nil || !errors.Is(err, os.ErrInvalid) {
+			return err
+		}
+
+		// Note: fs.openfile does not support opening a directory.
+		// So there is currently no way to set the mode of a directory.
+		// That's a good thing, because it means that we don't have to do
+		// permissions checks on parent directories.
+		flags := r.AttrFlags()
+		attrs := r.Attributes()
 		file, err := fs.openfile(r.Filepath, sshFxfWrite)
 		if err != nil {
 			return err
 		}
 
-		if r.AttrFlags().Size {
-			return file.Truncate(int64(r.Attributes().Size))
+		if flags.Size {
+			if err = file.Truncate(int64(attrs.Size)); err != nil {
```

Comment:
Fair enough.

Though, I would argue that err is probably the only variable that is reasonable to shadow inside a function.  I've had too many bugs over the long years (in multiple languages) that were due to shadowing variables that way.

Reply:
I’ve seen far more bugs of using logically-nil errors treated as if they were non-nil, than bugs where err is improperly shadowed. Mostly, because `err` in particular is almost always `err == nil` outside of the first `if` block that handles it.

Reply:
Fair enough.  As I said, err is probably the only variable where this makes sense. :)